challenge_1/c/xoumi/ReverseString.c

```diff
@@ -0,0 +1,27 @@
+#include<stdio.h>
+void main()
+{
+    int i,j;
+    char *string,temp;
+    printf("Enter string: ");
+    //because scanf is scary
+    gets(string);
+
+    //Just a loop to count length instead of using strlen because this isn't scary
+    for(i=0;string[i]!='\0';i++);
```

Comment:
Indeed strlen basically scans until it hits the 'null terminator' so this code is equivalent to using strlen. Just as a side note, `gets` is also scary and can be used in stack overflow attacks: check out [this stackoverflow thread](http://stackoverflow.com/questions/3302255/c-scanf-vs-gets-vs-fgets) for a  slightly more detailed description.